src/workerd/api/tests/abortsignal-test.js

```diff
@@ -188,3 +251,257 @@ export const onabortPrototypeProperty = {
+    const unusedAc = new AbortController();
+    const signal = AbortSignal.any([AbortSignal.timeout(200), unusedAc.signal]);
+    const res = await env.RpcRemoteEnd.countToInfinity(signal);
+
+    // We don't care the exact value it got to, but at least 1 iteration should have happened
+    ok(res.counter >= 1);
+
+    // Make sure the reason was passed without being garbled
+    ok(res.reason instanceof DOMException);
+    strictEqual(res.reason.message, 'The operation was aborted due to timeout');
+
+    // Make sure an event was dispatched on the remote side
+    ok(res.onAbortWasFired);
+  },
+};
+
+export const rpcAbortSignalAnyOnRemoteEnd = {
+  async test(ctrl, env, ctx) {
+    const ac = new AbortController();
+    const resPromise = env.RpcRemoteEnd.countToInfinityWithTimeout(ac.signal);
+
+    // Wait an arbitrary amount of time, then use the AbortController to abort the remote end.
+    await scheduler.wait(200);
+    const expectedReason =
+      'our timeout triggered before the 1000ms timeout on the other side';
+    ac.abort(expectedReason);
+
+    const res = await resPromise;
+
+    // We don't care the exact value it got to, but at least 1 iteration should have happened
+    ok(res.counter >= 1);
+
+    // Make sure the reason was passed without being garbled
+    strictEqual(res.reason, expectedReason);
+
+    // Make sure an event was dispatched on the remote side
+    ok(res.onAbortWasFired);
+  },
+};
+
+export const rpcRequestSignal = {
+  async test(ctrl, env, ctx) {
+    // Construct a request holding an AbortSignal, and then send this request to the other side
+    const req = new Request('http://example.com', {
+      signal: AbortSignal.timeout(200),
+    });
+
+    const res = await env.RpcRemoteEnd.countToInfinityWithRequest(req);
+
+    // We don't care the exact value it got to, but at least 1 iteration should have happened
+    ok(res.counter >= 1);
+
+    // Make sure the reason was passed without being garbled
+    ok(res.reason instanceof DOMException);
+    strictEqual(res.reason.message, 'The operation was aborted due to timeout');
+
+    // Make sure an event was dispatched on the remote side
+    ok(res.onAbortWasFired);
+  },
+};
+
+export const rpcCrossRequestSignal = {
+  async test(ctrl, env, ctx) {
+    // Save an AbortController in the global scope
+    await env.RpcRemoteEnd.tryUsingGlobalAbortController();
+
+    // Try to use it again
+    await rejects(
+      async () => env.RpcRemoteEnd.tryUsingGlobalAbortController(),
+      {
+        name: 'Error',
+        message:
+          "Cannot perform I/O on behalf of a different request. I/O objects (such as streams, request/response bodies, and others) created in the context of one request handler cannot be accessed from a different request's handler. This is a limitation of Cloudflare Workers which allows us to improve overall performance. (I/O type: RefcountedCanceler)",
+      }
+    );
+  },
+};
+
+export const rpcRemoteCanIgnoreSignal = {
```

Comment:
@danlapid Re: our discussion this morning. I added a test showing that just passing an AbortSignal over RPC, then aborting it, won't cause anything to happen by itself on the other end, unless some kind of handler is registered.

Maybe this will help a bit towards convincing ourselves that re-enabling Request.signal won't break existing usage.